*: make more tests compatible with cluster virtualization

knz opened the pull request:
Epic: CRDB-18499
Informs #76378.
Prerequisite to #108762.

(I manually tested this works with #108762)

knz committed:
testserver: do not verify enterprise license

Prior to this patch, for test tenant servers to be automatically used
two conditions needed to hold:

- the CCL package(s) needed to be imported.
- an enterprise license needed to be set (usually via `ccl.TestingEnableEnterprise()`)

While we want these conditions to hold to start separate-process SQL
servers in production, they do not need to be verified in unit tests.

NB: as a result of merging this PR, numerous (recently introduced)
calls to `ccl.TestingEnableEnterprise()` in various `main_test.go`
files are now unnecessary. Cleaning them up can be pushed to a later
phase.

Release note: None

knz committed:
serverutils: allow overriding the tenant selection package-wide

This will make it possible to merge the change that enables tenant
selection everywhere before we investigate the test failures in detail.

Release note: None

knz committed:
*: temporarily opt certain test packages out of cluster virtualization

Some packages contain many tests that fail when run with a virtual
cluster active. Until we have time to investigate them in details,
opt them out.

Release note: None

knz committed:
server: move stats_test.go to application_api/

It's a better home. Also, since the tests in that file already work
with cluster virtualization, we want this file out of 'server' where
we'll introduce a blanket disable.

Release note: None

knz committed:
server/application_api: make more tests compatible with cluster virtualization

Release note: None

knz committed:
rangedesc: properly opt out of cluster virtualization in tests

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/109329)
<!-- Reviewable:end -->

knz committed:
cdcevent: make more tests compatible with cluster virtualization

Release note: None

knz committed:
tenantcostserver: opt out of automatic test tenant

Release note: None

knz committed:
cdceval: make tests work with cluster virtualization

Release note: None

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 3 files at r1, 6 of 6 files at r2, 1 of 1 files at r3, 5 of 5 files at r4, 1 of 1 files at r5, 6 of 6 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/109329#-:-NcY6h9nBQ5ecDmH5xGG:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @miretskiy)



<!-- Sent from Reviewable.io -->

yuzefovich committed:
sql: adjust all subpackages to work with test tenant

Some of the tests are properly fixes, others have skips associated with
them.

Release note: None

yuzefovich committed:
sql/catalog: adjust many tests to work with test tenant

Three tests are skipped explicitly plus on the order of 10 tests that
use `newLeaseTest` helper too.

Release note: None

knz committed:
catalog/catkv: make more tests work with cluster virtualization

Release note: None

knz committed:
spanconfigkvaccessor: improve some tests to work with cluster virtualization

This also simplifies the test code.

Release note: None

knz committed:
tenantsettingswatcher: properly opt tests out of cluster virtualization

Release note: None

knz committed:
security: opt a test out of cluster virtualization

Release note: None

knz committed:
clisqlclient: make tests compatible with cluster virtualization

Release note: None

knz committed:
workload/insights: make test compatible with cluster virtualization

Release note: None

knz committed:
workload/bank: make test compatible with cluster virtualization

Release note: None

knz committed:
kvserver/reports: small test fixup

Release note: None

knz committed:
kvserver/rangelog: opt test out of cluster virtualization

Release note: None

knz committed:
kvserver/batcheval: make more tests work with cluster virtualization

Release note: None

dt commented on pkg/kv/kvserver/batcheval/cmd_export_test.go:
I don't think this is correct: if we're moving the start key into a tenant span obtained from ts.Codec, we need to move the end key as well

knz replied:
How?

dt replied:
maybe codec.TenantSpan().EndKey

knz committed:
rangefeedcache: make TestCache work with cluster virtualization

Release note: None

knz committed:
kvclient: make a test work with cluster virtualization

Release note: None

knz committed:
cli: opt some tests out of cluster virtualization

Release note: None

knz committed:
server/storage_api: opt some tests out of cluster virtualization

Release note: None

knz committed:
securitytest: enable CLI tests with secondary tenants

Release note: None

knz committed:
cli/testutils: avoid panic on error

Release note: None

knz committed:
colflow: opt out of virtual cluster testing

Release note: None

knz committed:
spanconfigmanagers: make more tests work with virtual clusters

Release note: None

knz committed:
workload/{bank,insights}: skip with cluster virtualization

These fail non-deterministically.

Release note: None

knz committed:
autoconfig: opt a test out of cluster virtualization

Release note: None

knz committed:
settingswatcher: make more tests work with cluster virtualization

Release note: None

knz committed:
sql/scheduledlogging: opt a test out of cluster virtualization

Release note: None

knz committed:
rangefeed: make more tests work with cluster virtualization

Release note: None

knz committed:
tenantcapabilitieswatcher: opt out of auto test tenant

Release note: None

knz committed:
spanconfigkvsubscriber: enhance some tests

Release note: None

knz committed:
logcrash: opt a test out of cluster virtualization

Release note: None

knz committed:
protectedts/ptcache: make more tests work with cluster virtualization

Release note: None

knz committed:
cli: make more tests compatible with cluster virtualization

Release note: None

knz committed:
systemconfigwatcher: opt a test out of cluster virtualization

Release note: None

knz committed:
spanconfigsqlwatcher: make more tests work with virtual clusters

Release note: None

knz committed:
serverutils: de-emphasize "tenant" in error message

Release note: None

knz committed:
catalog/lease: opt more tests into cluster virtualization

Release note: None

knz committed:
serverutils: clarify more test scenarios in logs

Release note: None

knz committed:
catkv: simplify tests

Release note: None

knz committed:
server/application_api: re-enable a skipped test

Release note: None

knz committed:
base: better document some options

Release note: None

knz commented:
TFYR!

bors r=yuzefovich

knz commented:
bors r=yuzefovich

knz commented:
bors r=yuzefovich

knz committed:
*: temporarily opt certain test packages out of cluster virtualization

Some packages contain many tests that fail when run with a virtual
cluster active. Until we have time to investigate them in details,
opt them out.

Release note: None

knz commented:
bors r=yuzefovich
